sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer.py

```diff
@@ -139,15 +140,36 @@ def _message_received(self, message):
         self._last_received_event = event_data
         self._on_event_received(event_data)
 
+    def _open(self):
```

Comment:
Sanity checking that we have live tests (Unit tests would be better, and may be possible for the "open never happens" scenario because we never hit the UAMQP do_work in that case) that validate the scenarios this might impact, namely:
- Open takes a while to happen successfully
- Open NEVER happens successfully.

Reply:
client_ready() calls UAMQP connection.work(). It will either succeed, or raise an exception, at some time.